utils/contract-wrapper.ts

```diff
@@ -1,47 +1,43 @@
-import {PriceFeed} from "../typechain/PriceFeed";
-import {ContractInterface, ethers, Signer} from "ethers";
-import {PriceFeed__factory} from "../typechain/factories/PriceFeed__factory";
+import { ethers, Signer } from "ethers";
+import { PriceFeed__factory } from "../typechain/factories/PriceFeed__factory";
+
+// TODO: Refactor this file and add many comments to make it more clear
 
 const { getSignedPrice } = require("../utils/api-connector");
 
 export type PriceDataType = {
     symbols: string[],
     values: number[],
     timestamp: number
-}
+};
 
 export type SignedPriceDataType = {
     priceData: PriceDataType,
     signer: string,
     signature: string
-}
+};
 
-
-async function getPriceData(signer:Signer, dataProvider:string, asset?: string) {
-    let {priceData, signature} = await getSignedPrice(dataProvider, asset);
+async function getPriceData(signer: Signer, dataProvider: string, asset?: string) {
+    let { priceData, signature } = await getSignedPrice(dataProvider, asset);
 
     let priceFeed = PriceFeed__factory.connect(ethers.constants.AddressZero, signer);
     let setPriceTx = await priceFeed.populateTransaction.setPrices(priceData, signature);
     
-    let setPriceData = setPriceTx.data ? setPriceTx.data.substr(2) : "";
+    const setPriceData = remove0xFromHexString(setPriceTx.data, {
+      allowFalsyValues: true,
+    });
 
-    let clearPriceTx = await priceFeed.populateTransaction.clearPrices(priceData);
-    let clearPricePrefix = clearPriceTx.data ? clearPriceTx.data.substr(2,8) : "";
+    let clearPriceTx = await priceFeed.populateTransaction.clearPrices(priceData); // priceData may have any value, we don't use it
+    let clearPricePrefix = clearPriceTx.data ? clearPriceTx.data.substr(2,8) : ""; // we get 4 bytes from 2nd to 5th (skipping 1st (0x)) - this is the function signature clearPrices
```

Comment:
The comment could be betterr:
We skip two first characters ("0x") and get 8 next (4 bytes of signature encoded as HEX)
2nd and 5th could be misleading as we see (2,8)